core/bessctl.cc

```diff
@@ -595,57 +594,50 @@ class BESSControlImpl final : public BESSControl::Service {
 
   Status ListTcs(ServerContext*, const ListTcsRequest* request,
                  ListTcsResponse* response) override {
-    int wid_filter;
-    int i;
+    const bess::TrafficClass* target_root = nullptr;
 
-    wid_filter = request->wid();
-    if (wid_filter >= Worker::kMaxWorkers) {
+    int wid = request->wid();
+    if (wid >= Worker::kMaxWorkers) {
       return return_with_error(response, EINVAL,
                                "'wid' must be between 0 and %d",
                                Worker::kMaxWorkers - 1);
-    }
-
-    if (wid_filter < 0) {
-      i = 0;
-      wid_filter = Worker::kMaxWorkers - 1;
-    } else {
-      i = wid_filter;
-    }
-
-    for (; i <= wid_filter; i++) {
-      if (workers[i] == nullptr) {
-        continue;
-      }
-      bess::TrafficClass* root = workers[i]->scheduler()->root();
-      if (!root) {
-        continue;
-      }
-
-      root->Traverse([&response, i](bess::TCChildArgs* args) {
-        bess::TrafficClass* c = args->child();
-
-        ListTcsResponse_TrafficClassStatus* status =
-            response->add_classes_status();
-
-        collect_tc(c, i, status);
-
-        if (args->parent_type() == bess::POLICY_WEIGHTED_FAIR) {
-          auto ca = static_cast<bess::WeightedFairChildArgs*>(args);
-          status->mutable_class_()->set_share(ca->share());
-        } else if (args->parent_type() == bess::POLICY_PRIORITY) {
-          auto ca = static_cast<bess::PriorityChildArgs*>(args);
-          status->mutable_class_()->set_priority(ca->priority());
+    } else if (wid < 0) {
+      wid = -1;
```

Comment:
Minor: might it be worth changing -1 to something more descriptive (e.g. a constant / enum)?